src/hotspot/share/classfile/verifier.cpp

```diff
@@ -285,7 +286,9 @@ bool Verifier::is_eligible_for_verification(InstanceKlass* klass, bool should_ve
     // already been rewritten to contain constant pool cache indices,
     // which the verifier can't understand.
     // Shared classes shouldn't have stackmaps either.
-    !klass->is_shared() &&
+    // However, bytecodes for shared old classes can be verified because
+    // they have not been rewritten.
+    (!(klass->is_shared() && klass->is_rewritten())) &&
```

Comment:
the outmost brackets are not necessary. Could your remove it?

Reply:
Fixed. I also removed the extra include of metaspaceShared.hpp.